Inject `ApiCredentials` instead of reading them internally

mokagio opened the pull request:
That is, instead of doing a direct read under the hood. 

I realize that this makes little difference structurally, because all the files touched are part of Keystone, so it not like this change makes Keystone agnostic of `ApiCredentials`.

Also, the (current iteration of the) upcoming solution to replace `ApiCredentials` will use `BuildSettings.current` so we could have simply done a find and replace.

Still, structurally, I prefer to surface this kind of dependency where a type requires information from another at the method or init level. I don't like it when information from outside sources is accessed from within the implementation. It also makes testing more straightforward. In short, I consider it a good coding practice and I though I'd introduce it while I was working in this area.

Notice that I did not replace the Zendesk values reads because they are done in a way that would have required rewriting part of `ZendeskUtils` and that seemed out of scope:

https://github.com/wordpress-mobile/WordPress-iOS/blob/8d6157cfeaec9b54dc47070eea101e006ce8aebe/WordPress/Classes/Utility/ZendeskUtils.swift#L451-L455

To test: Given the change is just bubbling values up from implementation to signatures, CI should suffice.

mokagio committed:
Inject `ApiCredentials` `.client` and `.secret`

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/308">25.9</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

mokagio committed:
Inject `ApiCredentials` Google client details

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

mokagio committed:
Inject `ApiCredentials` Tenor API key

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

mokagio committed:
Inject `ApiCredentials` Sentry DSN

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

mokagio committed:
Inject `ApiCredentials` `encryptedLogKey`

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

mokagio committed:
Inject `ApiCredentials` `docsBotId`

That's instead of doing a direct read under the hood.

The aim of this is to get to a place where Keystone is agnostic of
`ApiCredentials`.

mokagio commented on WordPress/Classes/Utility/Logging/EventLoggingDataProvider.swift:
```diff
@@ -10,20 +10,23 @@ struct EventLoggingDataProvider: EventLoggingDataSource {
     /// A block that returns all existing log files
     private let fetchLogFiles: LogFilesCallback?
 
+    let loggingAuthenticationToken: String
+    let loggingEncryptionKey: String
```
I removed the documentation comments when moving these two properties above the init because they are inherited from `EventLoggingDataSource`.

---

P.S. I'm surprised the verb to use is "inherited" when the property are defined in the protocol, I would have used something like "required", but that's what Xcode uses:

<img alt="image" src="https://github.com/user-attachments/assets/d1193f86-e4b0-48a0-a146-c137d64b78ed" width=300/>

mokagio committed:
Merge branch 'trunk' into mokagio/separate-apicredentials-keystone
